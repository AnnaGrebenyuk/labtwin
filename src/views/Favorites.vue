<template>
  <div>
    <b-container>
      <b-row align-h="center">
        <b-pagination v-model="currentPage"
                      :total-rows="favoriteEvents.length"
                      :per-page="page.size"/>
      </b-row>
    </b-container>
    <events-list :events="displayedEvents" :favoritesMode="true" />
  </div>
</template>

<script>
import { mapState } from 'vuex';
import { BContainer, BRow, BPagination } from 'bootstrap-vue';
import EventsList from '../components/EventsList.vue';

export default {
  name: 'favorites',
  components: {
    EventsList, BContainer, BRow, BPagination,
  },
  data: () => ({
    currentPage: 1,
  }),
  computed: {
    ...mapState(['favoriteEvents', 'page']),
    numberOfPages() {
      return Math.ceil(this.favoriteEvents.length / this.page.size);
    },
    displayedEvents() {
      return this.favoriteEvents.slice((this.currentPage - 1) * this.page.size,
        (this.currentPage * this.page.size) - 1);
    },
  },
  methods: {
  },
};
</script>
