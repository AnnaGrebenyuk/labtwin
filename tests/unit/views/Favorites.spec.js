import {
  createLocalVue,
  shallowMount,
} from '@vue/test-utils';
import Vuex from 'vuex';
import Favorites from '@/views/Favorites.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Favorites View', () => {
  let store;
  let state;
  let wrapper;
  beforeAll(() => {
    state = {
      favoriteEvents: new Array(16),
      page: {
        size: 15,
      },
    };
    store = new Vuex.Store({
      state,
    });
  });

  it('renders correctly', () => {
    wrapper = shallowMount(Favorites, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders the correct number of pages', () => {
    expect(wrapper.vm.numberOfPages).toEqual(2);
  });
});
