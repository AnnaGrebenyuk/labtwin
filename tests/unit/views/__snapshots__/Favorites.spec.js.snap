// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites View renders correctly 1`] = `
<div>
  <b-container-stub
    tag="div"
  >
    <b-row-stub
      alignh="center"
      tag="div"
    >
      <b-pagination-stub
        align="left"
        arialabel="Pagination"
        ellipsistext="…"
        firsttext="«"
        labelfirstpage="Go to first page"
        labellastpage="Go to last page"
        labelnextpage="Go to next page"
        labelpage="Go to page"
        labelprevpage="Go to previous page"
        lasttext="»"
        limit="5"
        nexttext="›"
        perpage="15"
        prevtext="‹"
        totalrows="16"
        value="1"
      />
    </b-row-stub>
  </b-container-stub>
   
  <events-list-stub
    events=",,,,,,,,,,,,,"
    favoritesmode="true"
  />
</div>
`;
